nucypher/policy/conditions/context.py

```diff
@@ -94,6 +96,21 @@ def _resolve_user_address(
     return expected_address
 
 
+def _resolve_null_address(
+    providers: Optional[ConditionProviderManager] = None,
+    **context,
+) -> ChecksumAddress:
+    """
+    Returns the null address (0x0000000000000000000000000000000000000000).
+
+    This is a protected context variable that doesn't require any input data.
+
+    Note: The `providers` and `**context` parameters are unused but maintained
+    for interface consistency with other resolver functions in `_DIRECTIVES`.
+    """
+    return to_checksum_address(NULL_ADDRESS)
```

Comment:
You can gain some efficiency by just doing the following since the null address is already checksummed (nothing to checksum 😄 ):

```suggestion
    return ChecksumAddress(NULL_ADDRESS)
```